pkg/kubelet/kuberuntime/helpers_test.go

```diff
@@ -39,6 +39,64 @@ func seccompLocalhostPath(profileName string) string {
 	return "localhost/" + seccompLocalhostRef(profileName)
 }
 
+func TestIsInitContainerFailed(t *testing.T) {
+	tests := []struct {
+		status      *kubecontainer.Status
+		isFailed    bool
+		description string
+	}{
+		{
+			status: &kubecontainer.Status{
+				State:    kubecontainer.ContainerStateExited,
+				ExitCode: 1,
+			},
+			isFailed:    true,
+			description: "Init container which state is exited and exitcode is non-zero shoud return true",
```

Comment:
I left nits on this PR https://github.com/kubernetes/kubernetes/pull/104651. Could you incorporate those?